src/framework/results/data/average_data.hpp

```diff
@@ -82,39 +93,48 @@ class AverageData {
 
   // Number of datum that have been accumulated
   size_t count_ = 0;
+
+  // Flag for whether the data has been normalized by the counts
+  // to conver the accum and accum squared to mean and variance
+  // Once it is normalized no new data can be added
+  bool normalized_ = false;
 };
 
 //------------------------------------------------------------------------------
 // Implementation
 //------------------------------------------------------------------------------
-
 template <typename T>
-T AverageData<T>::mean() const {
-  return (count_ > 1) ? Linalg::div(T(accum_), double(count_)) : accum_;
+void AverageData<T>::normalize() {
+  if (!normalized_ && count_ >= 1) {
+    if (count_ > 1) {
```

Comment:
If we move the `mean` and `variance` results, subsequent calls to those functions will yield wrong results. Probably we don't need to call them again but if that's the case we may need to implement something to recompute the values, as `normalize` won't be executed again as `normalized_` will be `true`. What do you think?

Reply:
It should be fine to call mean and variance multiple times, only the first call will result in normalizing the data (this is because `normalize` does nothing if `normalized_ = true`). But if you move them then yes, you've destroyed the container. I think this is like any other C++ object, after you used move you shouldn't use it again. If you want to use it again you should copy or reference the data rather than move, however i don't think this situation comes up in our current use case.

Reply:
I think the movement is a bit more hidden because we are moving specific members instead of the whole object, but as we're not gonna reuse them it seems we are ok.